core/watcher/watchertest/watcher.go

```diff
@@ -21,7 +21,7 @@ type WatcherAssert[T any] func(c *gc.C, changes []T) bool
 // received at least the given changes.
 func SliceAssert[T any](expect ...T) WatcherAssert[T] {
 	return func(c *gc.C, changes []T) bool {
-		if len(changes) >= len(expect) {
+		if len(expect) >= len(changes) {
```

Comment:
This doesn't seem correct. How I read it:

> If the number of expectations exceeds the changes we received, check the length. 

This feels like you'd miss any changes that are larger than the expectation.

Reply:
Yes; this would pass if only _some_ of your expectations were observed.

Reply:
Ops, a last minute push failure. Thanks for picking this up!